### PR TITLE
Pre script

### PR DESCRIPTION
**Pull request by oliverwoodings**
Initial foray into supporting the smarserve pre script.

I had to fuck around a bit with commander in order to get sub commands working in a decent way. It looks a bit mental, but it works!

### Todo

- [x] Command: pull
- [x] Command: push
- [x] Command: revisions/log/history
- [x] Command: diff
- [x] Command: publish
- [x] Ensure stuff is feature flagged upstream
- [x] Make the CLI help text work with pre stuff
- [x] Audit changes being made

### CODE REVIEW

File: src/lib/fetch.js [hunk 1]
```diff
@@ -47,7 +47,9 @@ function fetchWithAuth (method) {
     function errorHandler (error) {
       const resp = error.response
       checkErrors(resp)
-      throw new Error(error)
+      const newError = new Error(error)
+      newError.originalError = error
+      throw newError
```
Comment (alanclarke) [inline src/lib/fetch.js]: can we just `throw error?`

**Comment by alanclarke** [comment]
looking great so far!

File: src/lib/fetch.js (hunk 1 above)
Comment (oliverwoodings) [reply src/lib/fetch.js]: I think it is done this way so that the stack trace makes more sense (points to this file, rather than something deep in axios)

**Comment by oliverwoodings** [comment]
@alanclarke @samuelfullerthomas this is ready for review

**Comment by oliverwoodings** [comment]
@alanclarke @samuelfullerthomas either of you got time for a review in the next couple of days please?

**Pull request merged**